### PR TITLE
cleanup template construction in code generator

### PR DESCRIPTION
**Pull request by jaypipes**
Major cleanup of the template construction technique used in the code
generator. I've deleted the entire `pkg/template` package (and
subpackages), combining all the duplicative template construction and
template variable structs into a common set of methods on the
`pkg/generate.Generator` struct. These new common methods are in
`pkg/generate/template.go` now.

All of the code in the `cmd/ack-generate/commands/controller.go` and
`cmd/ack-generate/commands/apis.go` files that previously manually
constructed Template objects is now gone, replaced with much shorter and
simpler calls to the `Generator`.

Issue #252 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jaypipes** [comment]
+546 −1,372

\o/

**Review by A-Hilaly — APPROVED** [review]
/lgtm 👍

File: pkg/generate/template.go [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	ttpl "text/template"
+
+	"github.com/aws/aws-controllers-k8s/pkg/model"
+	ackmodel "github.com/aws/aws-controllers-k8s/pkg/model"
```
[the remark this answers is not in the document]
Comment (jaypipes) [reply pkg/generate/template.go:25]: ah, yeah. I will correct.

[the remark this answers is not in the document]
Comment (jaypipes) [reply pkg/generate/template.go]: ok, fixed.

**Pull request merged**